src/Network/Email/Email.php

```diff
@@ -1885,4 +1890,78 @@ protected function _getContentTypeCharset()
         }
         return strtoupper($this->charset);
     }
+
+    /**
+     * Serializes the email object to a value that can be natively serialized and re-used
+     * to clone this email instance.
+     *
+     * It has certain limitations for viewVars that are good to know:
+     *
+     *    - ORM\Query executed and stored as resultset
+     *    - File resources read and stored as contents
+     *    - SimpleXmlElements stored as associative array
+     *    - Exceptions stored as strings
+     *    - Closures and PDOs are not supported.
+     *
+     * @return array Serializable array of configuration properties.
+     * @throws \Exception When a view var object can not be properly serialized.
+     */
+    public function jsonSerialize()
+    {
+        $properties = [
+            '_to', '_from', '_sender', '_replyTo', '_cc', '_bcc', '_subject', '_returnPath', '_readReceipt',
+            '_template', '_layout', '_viewRender', '_viewVars', '_theme', '_helpers', '_emailFormat',
+            '_emailPattern', '_attachments', '_domain', '_messageId', '_headers', 'charset', 'headerCharset',
+        ];
+
+        $array = [];
+
+        foreach ($properties as $property) {
+            $array[$property] = $this->{$property};
+        }
+
+        // Iterate through view vars to clean up and normalize
+        array_walk_recursive($array['_viewVars'], function (&$item, $key) {
+            // Store file contents instead of resource.
+            if (is_resource($item) && get_resource_type($item) == 'stream') {
+                $item = stream_get_contents($item);
+            }
```

Comment:
Oh for some reason I thought attachments weren't read until the email was prepared to be sent. Good to know that is not the case.